packages/braid-design-system/src/lib/components/private/AvoidWidowIcon/AvoidWidowIcon.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import type { Space } from '../../../css/atoms/atoms';
+import type { BoxProps } from '../../Box/Box';
+import { Box } from '../../Box/Box';
+
+import * as styles from './AvoidWidowIcon.css';
+
+interface Props extends Pick<BoxProps, 'children' | 'className'> {
+  iconPosition: 'leading' | 'trailing';
+  space?: Space;
+}
+
+export const AvoidWidowIcon = ({
```

Comment:
I'll batch a few suggestions into one and you can pick and choose what you like:

- make the `children` prop more restrictive
- rename the `children` prop to `icon` inside the render function, to make it clear (to humans) what it is
- move the comment above the HTML entity


```tsx
import type { UseIconProps } from '../../../hooks/useIcon';
import React from 'react';
import type { Space } from '../../../css/atoms/atoms';
import type { BoxProps } from '../../Box/Box';
import { Box } from '../../Box/Box';

import * as styles from './AvoidWidowIcon.css';

interface Props extends Pick<BoxProps, 'className'> {
  iconPosition: 'leading' | 'trailing';
  space?: Space;
  children: React.ReactElement<UseIconProps>;
}

export const AvoidWidowIcon = ({
  children: icon,
  iconPosition,
  space = 'xxsmall',
  className,
}: Props) => (
  <Box
    component="span"
    paddingRight={iconPosition === 'leading' ? space : undefined}
    paddingLeft={iconPosition === 'trailing' ? space : undefined}
    className={[styles.nowrap, className]}
    aria-hidden
  >
    {iconPosition === 'leading' ? icon : undefined}
    {/* Word joiner character, a zero-width non-breaking character to prevent the icon wrapping onto its own line */}
    &#8288;
    {iconPosition === 'trailing' ? icon : undefined}
  </Box>
);
```